diff --git a/var/spack/repos/builtin/packages/starpu/package.py b/var/spack/repos/builtin/packages/starpu/package.py
index 45a2ca2cd2..59be852968 100644
--- a/var/spack/repos/builtin/packages/starpu/package.py
+++ b/var/spack/repos/builtin/packages/starpu/package.py
@@ -139,6 +139,7 @@ def configure_args(self):
                 "--disable-mpi" if "~mpi" in spec else "--enable-mpi",
                 "--without-mpicc" if "~mpi" in spec else "--with-mpicc=%s" % mpicc,
                 "--with-hwloc=%s" % spec["hwloc"].prefix,
+                "--enable-maxcpus=24",
             ]
         )
 
